### ISSUE TITLE
Carbon Compose Modal accessibility violation

### ISSUE THREAD

**Issue by bryan-pegg23** [carbon-design-system/carbon#6543]
## Detailed description

experiencing same/similar issue that was in ticket https://github.com/carbon-design-system/carbon/issues/4587. `An interactive element/widget must have an accessible name.` violation is thrown on top level of compose modal component. can be seen in both sample application below and in story book here https://react.carbondesignsystem.com/?path=/story/composedmodal--example-usage-with-trigger-button

> What version of the Carbon Design System are you using?

 - versions, also found in codesandbox: 
    "carbon-components": "10.16.0",
    "carbon-components-react": "7.16.0",
    "carbon-icons": "7.0.7",
    "react": "16.3.1",
    "react-dom": "16.3.1",

> What did you expect to happen?

- no accessibility violations

> What WCAG 2.1 checkpoint does the issue violate?

https://aat.w3ibm.mybluemix.net/token/4950d599-abf0-4c72-b249-9eed7e3bf02e/82ec2f72-e53b-4c22-98b3-faab6bd311db/archives/2020JanDeploy//doc/w3/help/en-US/idhi_accessibility_check_g1156.html

> Please create a reduced test case in CodeSandbox
https://codesandbox.io/s/carbon-components-react-du1ip?file=/src/index.js
## Additional information

<img width="1465" alt="Screen Shot 2020-07-24 at 9 21 44 AM" src="https://user-images.githubusercontent.com/19974517/88395475-21d75900-cd8f-11ea-9365-c4f9dafa5887.png">

**Comment by emyarod** [comment]
the error doesn't seem to be present on https://carbon-components-react.netlify.com/?path=/story/composedmodal--example-usage-with-trigger-button which is the latest version of the component

**Comment by dakahn** [comment]
I'm seeing this error using Accessibility Checker with latest ruleset.
![image.png](https://images.zenhubusercontent.com/5bbe25039f6acf2a5a05a50f/b1db8e38-0655-408d-a5df-f99d7145448e)

The modal needs an accessible name or title of some kind as seen in [this WAI-Aria example](https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html)
![image.png](https://images.zenhubusercontent.com/5bbe25039f6acf2a5a05a50f/20fc2f37-1f4c-414c-a284-a6b24eafc7a1)

**Comment by dhyliu** [comment]
@dakahn Hello! This issue was opened around 4 months ago. Could I have your support to close it asap? It is now a blocker for several offerings on IBM Cloud to GA launch.
